---
title: "Holiday Movies Analysis"
author: "Saya Kondo"
date: "2025-01-22"
categories: [Python, Analysis] # tags for a blog post (e.g., python)
image: "Holiday_Movie.jpg"

toc: true
---

In [ ]:
import pandas as pd
from io import StringIO
from datetime import datetime, timezone
import time
import random

In [ ]:
holiday_movies = pd.read_csv(
    "https://bcdanl.github.io/data/holiday_movies.csv"
)
display(holiday_movies)

,tconst,title_type,primary_title,simple_title,year,runtime_minutes,average_rating,num_votes
0,tt0020356,movie,Sailor's Holiday,sailors holiday,1929,58.0,5.4,55
1,tt0020823,movie,The Devil's Holiday,the devils holiday,1930,80.0,6.0,242
2,tt0020985,movie,Holiday,holiday,1930,91.0,6.3,638
3,tt0021268,movie,Holiday of St. Jorgen,holiday of st jorgen,1930,83.0,7.4,256
4,tt0021377,movie,Sin Takes a Holiday,sin takes a holiday,1930,81.0,6.1,740
...,...,...,...,...,...,...,...,...
2260,tt9747440,tvMovie,A Christmas Love Story,a christmas love story,2019,84.0,6.9,1652
2261,tt9747450,tvMovie,Holiday for Heroes,holiday for heroes,2019,81.0,7.0,1655
2262,tt9802890,tvMovie,Christmas Jars,christmas jars,2019,93.0,7.3,914
2263,tt9815084,tvMovie,A Very British Christmas,a very british christmas,2019,90.0,5.7,725


In [ ]:
holiday_movies.info

<bound method DataFrame.info of          tconst title_type             primary_title  \
0     tt0020356      movie          Sailor's Holiday   
1     tt0020823      movie       The Devil's Holiday   
2     tt0020985      movie                   Holiday   
3     tt0021268      movie     Holiday of St. Jorgen   
4     tt0021377      movie       Sin Takes a Holiday   
...         ...        ...                       ...   
2260  tt9747440    tvMovie    A Christmas Love Story   
2261  tt9747450    tvMovie        Holiday for Heroes   
2262  tt9802890    tvMovie            Christmas Jars   
2263  tt9815084    tvMovie  A Very British Christmas   
2264  tt9892854    tvMovie                     #Xmas   

                  simple_title  year  runtime_minutes  average_rating  \
0              sailors holiday  1929             58.0             5.4   
1           the devils holiday  1930             80.0             6.0   
2                      holiday  1930             91.0             6.3   
3         holiday of st jorgen  1930             83.0             7.4   
4          sin takes a holiday  1930             81.0             6.1   
...                        ...   ...              ...             ...   
2260    a christmas love story  2019             84.0             6.9   
2261        holiday for heroes  2019             81.0             7.0   
2262            christmas jars  2019             93.0             7.3   
2263  a very british christmas  2019             90.0             5.7   
2264                      xmas  2022             84.0             5.8   

      num_votes  
0            55  
1           242  
2           638  
3           256  
4           740  
...         ...  
2260       1652  
2261       1655  
2262        914  
2263        725  
2264        926  

[2265 rows x 8 columns]>

In [ ]:
holiday_movie_genres = pd.read_csv(
    "https://bcdanl.github.io/data/holiday_movie_genres.csv"
)
display(holiday_movie_genres)

,tconst,genres
0,tt0020356,Comedy
1,tt0020823,Drama
2,tt0020823,Romance
3,tt0020985,Comedy
4,tt0020985,Drama
...,...,...
4526,tt9815084,Family
4527,tt9815084,Romance
4528,tt9892854,Comedy
4529,tt9892854,Drama


In [ ]:
holiday_movie_genres.info

<bound method DataFrame.info of          tconst   genres
0     tt0020356   Comedy
1     tt0020823    Drama
2     tt0020823  Romance
3     tt0020985   Comedy
4     tt0020985    Drama
...         ...      ...
4526  tt9815084   Family
4527  tt9815084  Romance
4528  tt9892854   Comedy
4529  tt9892854    Drama
4530  tt9892854  Romance

[4531 rows x 2 columns]>

##Counting

In [ ]:
holiday_movies.shape

(2265, 8)

This dataset contains 2265 movies that are holiday-themed.

## Sorting

In [ ]:
top_rated = (
    holiday_movies
    .sort_values("average_rating", ascending=False)
    .head(10)
)

top_rated[["primary_title", "year", "average_rating", "num_votes"]]


,primary_title,year,average_rating,num_votes
1204,NLO Spirit of Christmas,2009,10.0,18
1433,Bringing Back Christmas,2023,9.9,20
1522,Christmas Bone Us,2012,9.8,10
1601,Cheap vs. Expensive Xmas Day,2022,9.5,14
519,Marie Osmond's Merry Christmas,1986,9.4,23
564,Christmas and the Civil War,2006,9.3,16
1416,Aunty Donna: Always Room for Christmas Pud,2018,9.3,24
2241,D'Astrain No Cinema - Christmas Special,2018,9.3,81
548,Tom's Christmas Tree,2006,9.2,34
1645,Christmas Ringer,2023,9.2,21


By sorting the data, the movie that has the highest average rating is "NLO Spirit of Christmas" with 10.0. Interestingly, the top ten highest average rating movies all contain a word "Christmas".

In [ ]:
top_rated_ = (
    holiday_movies
    .sort_values("average_rating", ascending=False)
    .head(10)
)

top_rated[["primary_title", "year", "average_rating", "num_votes"]]

,primary_title,year,average_rating,num_votes
1204,NLO Spirit of Christmas,2009,10.0,18
1433,Bringing Back Christmas,2023,9.9,20
1522,Christmas Bone Us,2012,9.8,10
1601,Cheap vs. Expensive Xmas Day,2022,9.5,14
519,Marie Osmond's Merry Christmas,1986,9.4,23
564,Christmas and the Civil War,2006,9.3,16
1416,Aunty Donna: Always Room for Christmas Pud,2018,9.3,24
2241,D'Astrain No Cinema - Christmas Special,2018,9.3,81
548,Tom's Christmas Tree,2006,9.2,34
1645,Christmas Ringer,2023,9.2,21


##Filtering

In [ ]:
recent_movies = holiday_movies[holiday_movies["year"] >= 2000]
recent_movies.shape

(1789, 8)

More than half of movies in the data set are released after year 2000.

##Joining & Counting

In [ ]:
holiday_full = pd.merge(
    holiday_movies,
    holiday_movie_genres,
    on="tconst",
    how="left"
)
holiday_full

,tconst,title_type,primary_title,simple_title,year,runtime_minutes,average_rating,num_votes,genres
0,tt0020356,movie,Sailor's Holiday,sailors holiday,1929,58.0,5.4,55,Comedy
1,tt0020823,movie,The Devil's Holiday,the devils holiday,1930,80.0,6.0,242,Drama
2,tt0020823,movie,The Devil's Holiday,the devils holiday,1930,80.0,6.0,242,Romance
3,tt0020985,movie,Holiday,holiday,1930,91.0,6.3,638,Comedy
4,tt0020985,movie,Holiday,holiday,1930,91.0,6.3,638,Drama
...,...,...,...,...,...,...,...,...,...
4526,tt9815084,tvMovie,A Very British Christmas,a very british christmas,2019,90.0,5.7,725,Family
4527,tt9815084,tvMovie,A Very British Christmas,a very british christmas,2019,90.0,5.7,725,Romance
4528,tt9892854,tvMovie,#Xmas,xmas,2022,84.0,5.8,926,Comedy
4529,tt9892854,tvMovie,#Xmas,xmas,2022,84.0,5.8,926,Drama


In [ ]:
genre_counts = (
    holiday_full["genres"]
    .value_counts()
)

genre_counts


,count
genres,
Comedy,1025
Drama,828
Romance,737
Family,707
Animation,268
Fantasy,185
Adventure,117
Documentary,101
Short,96


Comedy, drama, romance and Family dominate the genre distribution. This might be suggesting that holiday-themed movies primarily emphasize humor, emotion, and family-oriented storytelling rather than action or news.

##Indexing

In [ ]:
classic_movies = holiday_movies.loc[
    holiday_movies["year"] < 2000,
    ["primary_title", "year", "average_rating"]
]

classic_movies.sort_values("average_rating", ascending=False)


,primary_title,year,average_rating
519,Marie Osmond's Merry Christmas,1986,9.4
489,Morecambe & Wise Christmas Cracker,1993,9.0
199,John Grin's Christmas,1986,8.9
129,Julie Andrews: The Sound of Christmas,1987,8.9
1424,Christmas at River Cottage,1999,8.8
...,...,...,...
1885,Ms. Velma's Most Incredibly Magnificent Christ...,1976,2.2
238,The Star Wars Holiday Special,1978,2.1
212,Santa's Christmas Circus,1966,2.0
2100,The Christmas Light,1995,1.7


A movie that had the highest average rating by the year 2000 was Marie Osmond's Merry Christmas on 1986.

In [ ]:
classic_movies.sort_values("year", ascending=False)

,primary_title,year,average_rating
1424,Christmas at River Cottage,1999,8.8
325,Christmas Mission,1999,6.4
563,Snowden's Christmas,1999,6.7
331,O' Christmas Tree,1999,6.2
252,In the Christmas Spirit,1999,5.3
...,...,...,...
5,Sinners' Holiday,1930,6.3
2,Holiday,1930,6.3
3,Holiday of St. Jorgen,1930,7.4
1,The Devil's Holiday,1930,6.0


The oldest holiday themed movie was released in 1929 called Sailor's Holiday.